tests/ubpf_execution.rs

```diff
@@ -245,6 +242,59 @@ fn test_alu64_arithmetic() {
     );
 }
 
+#[test]
```

Comment:
What led you to create this test?

Reply:
In bpf programs there is a "128bit" test and the line `assert_eq!(solana_bpf_rust_128bit_dep::umultiply(30, 20), 600u128);` failed, so I disassembled it and turned it into a unit test. As it turned out, RCX was always saved and restored in the multiplication, even if it was the destination. So every multiplication with r3 as dst was wrong.